### PR TITLE
CVE-2017-12615

### PR DESCRIPTION
**Pull request by GORosin**
Filled out YAML file with my findings and comments. Log contains commits from VCC to fix

### CODE REVIEW

File: cves/CVE-2009-3548.yml [hunk 1]
```diff
@@ -85,7 +92,7 @@ upvotes_instructions: |
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes:
+upvotes: 4
```
Comment (mxm8856) [inline cves/CVE-2009-3548.yml]: upvotes: 6

File: cves/CVE-2009-3548.yml [hunk 2]
```diff
@@ -99,9 +106,9 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
-  code:
-  fix:
+  answer: No, only one file changed for fix, and this didn't involve unit tests.
+  code: false
+  fix: false
```
Comment (mxm8856) [inline cves/CVE-2009-3548.yml]: Which file? Do you think this fix could be unit tested?

File: cves/CVE-2009-3548.yml [hunk 3]
```diff
@@ -116,9 +123,14 @@ discovered:
...
-  date:
-  automated:
+  answer: |
+    This issue was reported directly to the tomcat users public mailing
+    list by David Horheim. Link - 
+    https://markmail.org/thread/wfu4nff5chvkb6xp#query:+page:1+mid:bva3jqihjxscatev+state:results. -
+    He mentions that while downloading exe version of the installer (specifically not zip or tar.gz)
+    it enables the manager application - which leads to the default blank password.
+  date: 2009-10-26
+  automated: false
```
Comment (mxm8856) [inline cves/CVE-2009-3548.yml]: Great explanation. Thanks for linking the resource.

File: cves/CVE-2009-3548.yml [hunk 4]
```diff
@@ -139,12 +151,13 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this
     section by explaining what happened between the VCCs and the fix.
-  answer:
+  answer: |
+    Most of the commits are related to bug fixes with links to the bugs (9 in total). Others pertain to changing
+    defaults. The interesting thing here is that the original code written by rmaucher seems to have caused a lot
+    of issues. Given, it does seem to be a pretty important config file so that is to be expected.
```
Comment (mxm8856) [inline cves/CVE-2009-3548.yml]: This comment shows a detailed analysis. I would love to see the commit numbers referenced so future readers can do their own analysis.

File: cves/CVE-2009-3548.yml [hunk 5]
```diff
@@ -183,8 +196,12 @@ lessons:
...
     note:
   secure_by_default:
-    applies:
-    note:
+    applies: true
+    note: |
+      By default, the admin password should at least be specified for some amount of security.
+      If a user just clicks through midlessly during installation, the blank password will persist.
+      This vulnerability defaulted it to blank. Perhaps this is not the user's main concern. Nonetheless,
+      not secure by default.
```
Comment (mxm8856) [inline cves/CVE-2009-3548.yml]: I absolutely agree with this. This should be a default requirement. I wonder if the developer missed the requirement or if this was a result of laziness.

File: cves/CVE-2017-12615.yml [hunk 6]
```diff
@@ -76,7 +79,7 @@ incomplete_fix_instructions: |
   Please list any fix commits for this vulnerability that had to be corrected
   at a later date.
 incomplete_fixes:
-  - commit:
+  - commit: e862b0c259b4c867553df29cef1af9e904af83b0
     note:
```
Comment (mxm8856) [inline cves/CVE-2017-12615.yml]: why was this fix not complete? Was there any commit message with this specific commit?

File: cves/CVE-2017-12615.yml [hunk 7]
```diff
@@ -101,9 +104,9 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
+  answer: When I checked the files that were changed there were no unit test files indicated
```
Comment (mxm8856) [inline cves/CVE-2017-12615.yml]: In your opinion, could this fix be resolved with a unit test?

File: cves/CVE-2017-12615.yml [hunk 8]
```diff
@@ -118,7 +121,10 @@ discovered:
     If there is no evidence as to how this vulnerability was found, then you
     may leave the entries blank except for "answer", BUT please write down
     where you looked in "answer".
-  answer:
+  answer: |
```
Comment (mxm8856) [inline cves/CVE-2017-12615.yml:124]: Were these inhouse tests or automated third-party?

File: cves/CVE-2017-12615.yml [hunk 9]
```diff
@@ -161,37 +171,41 @@ lessons:
...
-    applies:
+    applies: false
     note:
   distrust_input:
-    applies:
-    note:
+    applies: true
+    note: |
+     File input was missing some sanitization, as using a "/" would allow the malicious file 
+     to go through to the server.
```
Comment (mxm8856) [inline cves/CVE-2017-12615.yml:191]: Distrust input fits this vulnerability the best. All input should be sanitized. If that was the case, this vulnerability never would have existed.

File: cves/CVE-2017-12615.yml [hunk 10]
```diff
@@ -161,37 +171,41 @@ lessons:
...
-    applies:
+    applies: false
     note:
   environment_variables:
-    applies:
+    applies: false
     note:
   secure_by_default:
-    applies:
+    applies: false
```
Comment (mxm8856) [inline cves/CVE-2017-12615.yml]: I could argue this applies. It ties in with distrust input. Input never should be trusted so by default it should be in the system.

File: cves/CVE-2017-12615.yml [hunk 11]
```diff
@@ -204,4 +218,7 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: |
```
Comment (mxm8856) [inline cves/CVE-2017-12615.yml:221]: Was there any reference to a requirement? Were there any automated tests for this section of code?

File: cves/CVE-2017-12615.yml [hunk 12]
```diff
@@ -10,21 +10,21 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE:
+CWE: 650
```
Comment (mxm8856) [inline cves/CVE-2017-12615.yml]: upvotes: 3

File: cves/CVE-2017-12615.yml (hunk 6 above)
Comment (GORosin) [reply cves/CVE-2017-12615.yml]: From here: https://svn.apache.org/viewvc?view=revision&revision=1804604
The commit says that this is only part of the fix for this CVE. The fix was completed later in another revision (listed in fixes).

**Pull request merged**